project-koku/koku
# Validating source cost-model associations for cost-model creation/update

Throwing an exception when a cost-model update or create is attempted and a source on the `source_uuids` list is already associated with a cost model.

**Testing**
1. Create two OCP sources (Source 1 and Source 2).
2. Create a cost model (Cost Model 1) with Source 1.
3. Create a second cost model (Cost Model 2) with both OCP sources.  Verify creation fails.
4. Remove Source 1 from Cost Model 1 and attempt to create Cost Model 2 again.  Verify it's successful and both sources are associated.
5. Update Cost Model 1 with Source 1 and verify that it fails.
6. Remove Source 1 from Cost Model 2, then attempt to update Cost Model 1 with Source 1 again.  Verify it's successful.

**Test Results**
[cost_model_no_alter_ut.txt](https://github.com/project-koku/koku/files/4636070/cost_model_no_alter_ut.txt)
